include/aws/io/event_loop.h

```diff
@@ -140,6 +144,21 @@ struct aws_event_loop_group {
     struct aws_shutdown_callback_options shutdown_options;
 };
 
+typedef struct aws_event_loop *(aws_new_system_event_loop_fn)(struct aws_allocator *alloc,
+                                                              const struct aws_event_loop_options *options);
+
+struct aws_event_loop_configuration {
+    enum aws_event_loop_style style;
+    aws_new_system_event_loop_fn *event_loop_new_fn;
+    const char *name;
+    bool is_default;
+};
```

Comment:
Yes, the `is_default` is not functional for now, and is just reserved for a later usage. Currently, we use pre-compile definitions to determine which event-loop we would like to use in aws-c-io. In the future, we would like to allow a runtime configuration to set the event loop, so that the user could make choice theirselves. (Probably after we make sure the dispatch queue is in a stable status.) Once that's there, as we would have multiple event loop implementation enabled, the `is_default` would be used to set the "default" configuration. 
Let me add the comments to explain the details there.